tests/core/policies/test_rule_policy.py

```diff
@@ -1936,3 +1944,554 @@ def test_predict_nothing_if_fallback_disabled():
     )
 
     assert prediction.max_confidence == 0
+
+
+def test_hide_rule_turn():
+    chitchat = "chitchat"
+    action_chitchat = "action_chitchat"
+    domain = Domain.from_yaml(
+        f"""
+        intents:
+        - {GREET_INTENT_NAME}
+        - {chitchat}
+        actions:
+        - {UTTER_GREET_ACTION}
+        - {action_chitchat}
+    """
+    )
+    chitchat_story = TrackerWithCachedStates.from_events(
+        "chitchat story",
+        domain=domain,
+        slots=domain.slots,
+        evts=[
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": chitchat}),
+            ActionExecuted(action_chitchat),
+            ActionExecuted(ACTION_LISTEN_NAME),
+        ],
+    )
+    policy = RulePolicy()
+    policy.train(
+        [GREET_RULE, chitchat_story], domain, RegexInterpreter(),
+    )
+
+    conversation_events = [
+        ActionExecuted(ACTION_LISTEN_NAME),
+        UserUttered("haha", {"name": GREET_INTENT_NAME}),
+    ]
+    prediction = policy.predict_action_probabilities(
+        DialogueStateTracker.from_events(
+            "casd", evts=conversation_events, slots=domain.slots
+        ),
+        domain,
+        RegexInterpreter(),
+    )
+    assert_predicted_action(prediction, domain, UTTER_GREET_ACTION)
+    assert prediction.hide_rule_turn
+
+    conversation_events += [
+        ActionExecuted(UTTER_GREET_ACTION, hide_rule_turn=prediction.hide_rule_turn)
+    ]
+    prediction = policy.predict_action_probabilities(
+        DialogueStateTracker.from_events(
+            "casd", evts=conversation_events, slots=domain.slots
+        ),
+        domain,
+        RegexInterpreter(),
+    )
+    assert_predicted_action(prediction, domain, ACTION_LISTEN_NAME)
+    assert prediction.hide_rule_turn
+
+    conversation_events += [
+        ActionExecuted(ACTION_LISTEN_NAME, hide_rule_turn=prediction.hide_rule_turn),
+        UserUttered("haha", {"name": chitchat}),
+    ]
+    tracker = DialogueStateTracker.from_events(
+        "casd", evts=conversation_events, slots=domain.slots
+    )
+    states = tracker.past_states(domain, ignore_rule_only_turns=True)
+    assert states == [
+        {},
+        {
+            USER: {TEXT: "haha", INTENT: chitchat},
+            PREVIOUS_ACTION: {ACTION_NAME: ACTION_LISTEN_NAME},
+        },
+    ]
+
+
+def test_hide_rule_turn_with_slots():
+    some_action = "some_action"
+    some_other_action = "some_other_action"
+    some_intent = "some_intent"
+    some_other_intent = "some_other_intent"
+    some_slot = "some_slot"
```

Comment:
optional: Very verbose, but makes the test easier to read, what do you think? 
```suggestion
    slot_which_is_only_in_rule = "some_slot"
```